gst/datarepo/gstdatareposink.c

```diff
@@ -293,6 +299,33 @@ gst_data_repo_sink_write_others (GstDataRepoSink * sink, GstBuffer * buffer)
   return GST_FLOW_OK;
 }
 
+/**
+ * @brief Function to validate flexible tensors
+ */
+static gboolean
+gst_data_repo_sink_validate_flexible_tensors (GstBuffer * buffer)
+{
+  gboolean ret;
+  guint num_tensors, i;
+  GstMemory *mem;
+  GstTensorMetaInfo meta;
+
+  g_return_val_if_fail (buffer != NULL, FALSE);
+
+  num_tensors = gst_buffer_n_tensor (buffer);
+
+  for (i = 0; i < num_tensors; i++) {
+    mem = gst_tensor_buffer_get_nth_memory (buffer, i);
+    ret = gst_tensor_meta_info_parse_memory (&meta, mem);
+    gst_memory_unref (mem);
+    if (ret)
```

Comment:
This is weird (invalid case when 1st mem block is flexible but n-th tensor is invalid, this returns true.)

Reply:
If there is at least one flexible tensor in the buffer, this function should return true .

Reply:
in case of ```format=flexible```, how can I get ```num_tensor```?

Reply:
It will be same, use gst_buffer_n_tensor().
Note that, now we does not implement flexible tensor case of extra tensors.

Reply:
thank you.

Reply:
If ```format = flexible```, 
are all tensors in the buffer flexible?

Reply:
> If `format = flexible`, are all tensors in the buffer flexible?

Currently yes. If caps is flexible and memories in gst-buffer do not have tensor-meta (it means memory block for flexible type), it is error case.

Reply:
Thank you. 
To read flexible files in datareposrc, each gstbuffer can be distinguished by offset in JSON.
but, 
'gst_buffer_memory_append() 'can be used only when we know the number and size of flexible tensors in gstbuffer.
we need the starting offset of each flexible tensors in a gstbuffer.